src/components/application_manager/rpc_plugins/sdl_rpc_plugin/include/sdl_rpc_plugin/commands/hmi/get_device_connection_status_request.h

```diff
@@ -0,0 +1,163 @@
+#define SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_COMMANDS_HMI_GET_DEVICE_CONNECTION_STATUS_REQUEST_H_
+
+#include "application_manager/commands/request_from_hmi.h"
+
+namespace sdl_rpc_plugin {
+namespace app_mngr = application_manager;
+
+namespace commands {
+
+/**
+ * @brief GetDeviceConnectionStatusRequest command class
+ * Retrieve requested values at <device> param from
+ * "device_data" section of PolicyTable.
+ * If <device> param is empty all the <ID>, <name> of device,
+ * <usb_transport_status>, <transport_type> params are retrieved
+ * from "device_data" section of PolicyTable.
+ **/
+class GetDeviceConnectionStatusRequest
+    : public app_mngr::commands::RequestFromHMI {
+ public:
+  /**
+   * @brief GetDeviceConnectionStatusRequest class constructor
+   * @param message Incoming SmartObject message
+   **/
+  GetDeviceConnectionStatusRequest(
+      const app_mngr::commands::MessageSharedPtr& message,
+      app_mngr::ApplicationManager& application_manager,
+      app_mngr::rpc_service::RPCService& rpc_service,
+      app_mngr::HMICapabilities& hmi_capabilities,
+      policy::PolicyHandlerInterface& policy_handler);
+
+  /**
+   * @brief GetDeviceConnectionStatusRequest class destructor
+   **/
+  virtual ~GetDeviceConnectionStatusRequest();
+
+  /**
+   * @brief Execute command
+   **/
+  void Run() OVERRIDE;
+
+ private:
+  DISALLOW_COPY_AND_ASSIGN(GetDeviceConnectionStatusRequest);
+
+  /**
+   * @brief Reads the mandatory parameters
+   * as defined in APPLINK-28267
+   * On successfull
+   * @param device_data Corresponding to <device> parameter DeviceInfo
+   * structure.
+   * @param out_device_id On success the "id" of the device is stored here.
+   * @return True if the mandatory parameters are valid.
+   */
+  bool GetMandatoryParameters(const smart_objects::SmartObject& device_data,
+                              std::string& out_device_id) const;
+
+  /**
+   * @brief Reads the string value corresponding to key from device_data
+   * @param device_data Corresponding to <device> parameter DeviceInfo
+   * structure.
+   * @param key Key used to read the corresponding string value from device_data
+   * @param out_value Corresponding to key string value from device_data
+   * @param True if the out_value is valid
+   */
+  bool GetKeyStringValue(const smart_objects::SmartObject& device_data,
+                         const std::string& key,
+                         std::string& out_value) const;
+
+  /**
+   * @brief Check whether the string contains
+   * invalid characters only as defined in APPLINK-28280
+   */
+  bool IsSyntaxWrong(const std::string& str) const;
+};
+
+/**
+ * @brief Functor to retrieve all the data from the "device_data"
+ * section of the policy table and store <ID>, <name> of device,
+ * <usb_transport_status>, <transport_type> as array with key <device>
+ * in the message
+ */
+class DeviceDataRetriever {
```

Comment:
@ZhdanovP I think you could move this class under the private section of `GetDeviceConnectionStatusRequest` as this is just an implementation detail

Reply:
@AKalinich-Luxoft yes this class is already in private